libminifi/test/archive-tests/CompressContentTests.cpp

```diff
@@ -91,851 +91,835 @@ class ReadCallback: public org::apache::nifi::minifi::InputStreamCallback {
-    }
+  LogTestController::getInstance().setTrace<org::apache::nifi::minifi::processors::CompressContent>();
+  LogTestController::getInstance().setTrace<org::apache::nifi::minifi::processors::LogAttribute>();
+  LogTestController::getInstance().setTrace<core::ProcessSession>();
+  LogTestController::getInstance().setTrace<core::repository::VolatileContentRepository>();
+  // LogTestController::getInstance().setTrace<core::repository::FileSystemRepository>();
+  LogTestController::getInstance().setTrace<org::apache::nifi::minifi::Connection>();
+  LogTestController::getInstance().setTrace<org::apache::nifi::minifi::core::Connectable>();
+  LogTestController::getInstance().setTrace<org::apache::nifi::minifi::io::FileStream>();
+
+  std::shared_ptr<TestRepository> repo = std::make_shared<TestRepository>();
+
+  std::shared_ptr<core::Processor> processor = std::make_shared<org::apache::nifi::minifi::processors::CompressContent>("compresscontent");
+  std::shared_ptr<core::Processor> logAttributeProcessor = std::make_shared<org::apache::nifi::minifi::processors::LogAttribute>("logattribute");
+  processor->initialize();
+  utils::Identifier processoruuid;
+  REQUIRE(true == processor->getUUID(processoruuid));
+  utils::Identifier logAttributeuuid;
+  REQUIRE(true == logAttributeProcessor->getUUID(logAttributeuuid));
+
+  std::shared_ptr<core::ContentRepository> content_repo = std::make_shared<core::repository::VolatileContentRepository>();
+  // std::shared_ptr<core::ContentRepository> content_repo = std::make_shared<core::repository::FileSystemRepository>();
+  content_repo->initialize(std::make_shared<org::apache::nifi::minifi::Configure>());
+  // connection from compress processor to log attribute
+  std::shared_ptr<minifi::Connection> connection = std::make_shared<minifi::Connection>(repo, content_repo, "logattributeconnection");
+  connection->addRelationship(core::Relationship("success", "compress successful output"));
+  connection->setSource(processor);
+  connection->setDestination(logAttributeProcessor);
+  connection->setSourceUUID(processoruuid);
+  connection->setDestinationUUID(logAttributeuuid);
+  processor->addConnection(connection);
+  // connection to compress processor
+  std::shared_ptr<minifi::Connection> compressconnection = std::make_shared<minifi::Connection>(repo, content_repo, "compressconnection");
+  compressconnection->setDestination(processor);
+  compressconnection->setDestinationUUID(processoruuid);
+  processor->addConnection(compressconnection);
+
+  std::set<core::Relationship> autoTerminatedRelationships;
+  core::Relationship failure("failure", "");
+  autoTerminatedRelationships.insert(failure);
+  processor->setAutoTerminatedRelationships(autoTerminatedRelationships);
+
+  processor->incrementActiveTasks();
+  processor->setScheduledState(core::ScheduledState::RUNNING);
+  logAttributeProcessor->incrementActiveTasks();
+  logAttributeProcessor->setScheduledState(core::ScheduledState::RUNNING);
+
```

Comment:
Minor, but searching for this exact code block (including the line left commented out) results 8 matches.

It might improve readibility to refactor this instead of copy-pasting.

<img width="1086" alt="Screenshot 2020-06-05 at 10 47 20" src="https://user-images.githubusercontent.com/64011968/83856497-f1bdf300-a719-11ea-9dd9-1cd6636d5cf7.png">

Reply:
Most (all?) of this is unchanged old code that shows up in the diff only because it was reindented.

Reply:
yes it is, we could refactor it, but I would argue that it is out of scope for this issue

Reply:
I thought this was the rule?

<img width="623" alt="Screenshot 2020-06-05 at 13 52 16" src="https://user-images.githubusercontent.com/64011968/83873255-c516d500-a733-11ea-847b-99b9e16e6b65.png">

Reply:
done

Reply:
I think we only need to make things we touch "nice" to a sensible degree. To demonstrate another extreme, if one day we decide to change our indentation to tabs, and someone contributes the reindentation, he should not need to also clean up all questionable parts of the codebase.

A sensible degree is subjective, so I usually make things nice if they really bother me or when they are easy fixes.